micro-rdk/src/common/conn/viam.rs

```diff
@@ -558,7 +572,7 @@ where
             None => None,
         };
 
-        // TODO(RSDK-9887): remove after implementing agent NVS storage
+        // TODO(RSDK-10062): determine where additional network updates logic should live
```

Comment:
I want to align this better with how we do update the robot config but to do that properly we also need to do a check in the config update monitor for changes in the agent config.

added a ticket so I can come back to it after adding the network tickets.